### PR TITLE
feat: add Talos events sink proto files and the reference implementation

### PR DESCRIPTION
**Pull request by Unix4ever**
This PR adds protocol definitions for the events sink gRPC service.
The executable `events-sink` is a reference implementation fo the
service that simply logs all incoming events.

Signed-off-by: Artem Chernyshev <artem.chernyshev@talos-systems.com>

### CODE REVIEW

File: .kres.yaml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+---
+kind: common.Docker
+name: setup-ci
+spec:
+    allowInsecure: true
+---
+kind: golang.UnitTests
+name: unit-tests
+spec:
+    requiresInsecure: true
```
Comment (smira) [inline .kres.yaml:10]: do we need it??

Comment (Unix4ever) [reply .kres.yaml:10]: not really :D Copypaste issue

File: api/events/events.proto [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+message EventRequest {
+  google.protobuf.Any data = 1;
+  string id = 2;
+}
+
+message EventResponse {}
+
+service EventSinkService {
+  rpc Publish(stream EventRequest) returns(EventResponse);
```
Comment (smira) [inline api/events/events.proto:17]: my concern with streaming implementation here is that we would have no way to know that some events were delivered (from the Talos side). we just send the events and _hope_ that they got delivered

if we do a single-shot (unary) call, we have a response back immediately, and we know we should retry or we know that all events got pushed

Comment (Unix4ever) [reply api/events/events.proto]: yeah... makes sense. I've updated

**Review by smira — APPROVED** [review]

**Pull request merged**